### PR TITLE
github-cli: update to 2.4.0.

### PR DESCRIPTION
**Pull request by shizonic**
#### Testing the changes
- I tested the changes in this PR: **YES**

### CODE REVIEW

File: srcpkgs/github-cli/template [hunk 1]
```diff
@@ -1,19 +1,20 @@
...
 pkgname=github-cli
-version=2.0.0
+version=2.2.0
 revision=1
 wrksrc="cli-${version}"
 build_style=go
 build_helper=qemu
 go_import_path="github.com/cli/cli/cmd/gh"
 go_ldflags="-X github.com/cli/cli/internal/build.Version=v${version}"
+go_mod_mode="off"
```
Comment (ericonr) [inline srcpkgs/github-cli/template:10]: Not using Go modules is definitely a regression. Can you show where it fails?

Comment (shizonic) [reply srcpkgs/github-cli/template:10]: TBH: I don't know where execatly the problem is, but I get this error during building:

```shell
=> github-cli-2.2.0_1: building [go] [qemu] for x86_64...
   [host] go-1.17.2_1: found (https://alpha.de.repo.voidlinux.org/current)
=> github-cli-2.2.0_1: installing host dependencies: go-1.17.2_1 ...
=> github-cli-2.2.0_1: running pre-build hook: 02-script-wrapper ...
=> github-cli-2.2.0_1: running pre_build ...
=> github-cli-2.2.0_1: running do_build ...
no required module provides package github.com/cli/cli/cmd/gh; to add it:
	go get github.com/cli/cli/cmd/gh
=> ERROR: github-cli-2.2.0_1: do_build: 'go install -p "$XBPS_MAKEJOBS" -mod="${go_mod_mode}" -x -tags "${go_build_tags}" -ldflags "${go_ldflags}" ${go_package}' exited with 1
=> ERROR:   in do_build() at common/build-style/go.sh:36
```

After I added `go_mod_mode=off` it worked, so I pushed it.

**Comment by NyaaaWhatsUpDoc** [comment]
Might we want to build this using the repository makefile? Also the version info we're pulling in for the ldflags looks to be not using the v2 major version. See the build process in: https://github.com/cli/cli/blob/trunk/script/build.go

That might be the issue causing the above too.

**Comment by NyaaaWhatsUpDoc** [comment]
Making the following change fixed things in my experience:
```
diff --git a/srcpkgs/github-cli/template b/srcpkgs/github-cli/template
index 3044ccc38b..4b1251b900 100644
--- a/srcpkgs/github-cli/template
+++ b/srcpkgs/github-cli/template
@@ -5,9 +5,8 @@ revision=1
 wrksrc="cli-${version}"
 build_style=go
 build_helper=qemu
-go_import_path="github.com/cli/cli/cmd/gh"
-go_ldflags="-X github.com/cli/cli/internal/build.Version=v${version}"
-go_mod_mode="off"
+go_import_path="github.com/cli/cli/v2/cmd/gh"
+go_ldflags="-X github.com/cli/cli/v2/internal/build.Version=v${version}"
 short_desc="GitHub CLI tool"
 maintainer="shizonic <realtiaz@gmail.com>"
 license="MIT"
@@ -20,7 +19,7 @@ pre_build() {
        local _date
        if [ "$SOURCE_DATE_EPOCH" ]; then
                _date="$(date --utc --date "@$SOURCE_DATE_EPOCH" "+%Y-%m-%d")"
-               go_ldflags="$go_ldflags -X github.com/cli/cli/internal/build.Date=${_date}"
+               go_ldflags="$go_ldflags -X github.com/cli/cli/v2/internal/build.Date=${_date}"
        fi
 }
```

**Comment by shizonic** [comment]
Thanks. Fixed it and bumped it to 1.4.0.

**Comment by paper42** [comment]
@shizonic could you also fix update-check?

```sh
site="https://github.com/cli/cli/releases"
pattern='/tags/v\K[\d.]+(?=.tar.gz)'
```

**Comment by shizonic** [comment]
Done.

**Review by LaloQuera710 — COMMENTED** [review]
9163cd9371c00f75cc0ad63d3cb6657e4676826f

**Pull request merged**